sources/packages/api/src/route-controllers/student-appeal/models/student-appeal.dto.ts

```diff
@@ -0,0 +1,18 @@
+import { ArrayMinSize, IsArray } from "class-validator";
+/**
+ * DTO for student appeal request.
+ */
+export class StudentAppealRequestDTO {
+  formName: string;
+  formData: any;
+}
+
+/**
+ * DTO for student appeal.
+ */
+export class StudentAppealDTO {
+  applicationId?: number;
+  @IsArray()
+  @ArrayMinSize(1)
+  studentAppealRequests: StudentAppealRequestDTO[];
```

Comment:
Please add the validation to the child objects. Can you try adding the below?
```
@IsArray()
@ArrayMinSize(1)
@ValidateNested({ each: true })
@Type(() => <child type>)
```

Reply:
For the StudentAppealRequestDTO the dryRunSubmission is happening and it is happening in loop, that is the reason why I did not add class-validators to it's properties. But the dry run does not check if at least one object is present in array for StudentAppealDTO. Hence I added class validator only there.

Reply:
Not a proper reason to skip the validation here. The dry run is validating part of the DTO that, for instance, needs the _formName_ that must be present. Adding the validators here will avoid, for instance, a useless call for a _dryRun_.

Reply:
IMO dry definitely required because the formdata can be anything and is dynamic so we cannot do a perfect validation with class validators for formData. On the formname, why I did not do validation because formname is also a participant of dryrun and i believe unless it is perfect, dryrun will not be successful. Still ok to add class validator for formName but felt it can be skipped as well.

Reply:
Thanks for doing the change as we talked 😉